rancher/charts
# Add fleet grafana dashboards

## Issue: 

https://github.com/rancher/fleet/issues/2460
## Problem

For the existing Fleet metrics (and Prometheus configuration to scrape those), no Grafana Dashboards exist yet.
## Solution

Adds Grafana dashboards for visualizing Fleet metrics.

## Testing

- On a cluster with Rancher and a fleet version >= v0.10.0-rc13, install the rancher-monitoring chart that includes the changes of this PR.
- Open Grafana through the Rancher UI (section "monitoring") and look for 5 dashboards whose name starts with "Fleet". You will not find those dashboards embedded in the Rancher UI. ![image](https://github.com/rancher/charts/assets/9606095/8e42d8bf-68d8-4a90-94b7-781d0efeb4bc)
- You will need to create a GitRepo to start seeing GitRepo metrics.
- Check if all dashboards are showing "data". Note that, depending on the age of the monitoring installation, the time range may need to be decreased to start seeing visualizations in those dashboards. 
![image](https://github.com/rancher/charts/assets/9606095/3f0a1f59-80e8-42d9-9116-b8f328183958)



## Engineering Testing
### Manual Testing

See Testing.

### Automated Testing

Testing up to fetching and testing metrics from fleet-controller in https://github.com/rancher/fleet/tree/main/e2e/metrics. It does not cover the use of the ServiceMonitor to configure the Prometheus instance to scrape those metrics nor anything with Grafana.

## QA Testing Considerations

No additional testing required, tests as described in the Testing section are deemed sufficient.

### Regressions Considerations

I cannot think of any cases in which adding new Grafana dashboards would conflict with existing dashboards, except for the filenames perhaps. Chances of any regressions are low.

## Backporting considerations

Does not need to be backported.
